pandas/tests/strings/test_get_dummies.py

```diff
@@ -98,30 +96,77 @@ def test_get_dummies_with_pyarrow_dtype(any_string_dtype, dtype):
 
 
 # GH#47872
-@pytest.mark.xfail(using_string_dtype(), reason="TODO(infer_string)")
-def test_get_dummies_with_str_dtype(any_string_dtype):
+@pytest.mark.parametrize("use_string_repr", [True, False])
+def test_get_dummies_with_any_string_dtype(
+    request, any_string_dtype, any_string_dtype2, use_string_repr, using_infer_string
```

Comment:
I commented about this on the PR adding this test, see https://github.com/pandas-dev/pandas/pull/59577#issuecomment-2340442650, because I don't think it makes much sense to allow this. I see there is a PR following up on it to disallow `dtype=str` in `get_dummies` (https://github.com/pandas-dev/pandas/pull/59786), but we should follow-up on it to ensure it gets into 3.0

(but so conclusion: I would leave this alone for this PR, and focus on fixing it in https://github.com/pandas-dev/pandas/pull/59786)